floater/generators.py

```diff
@@ -105,19 +126,30 @@ def get_hexmesh(self):
         RETURNS
         -------
         x : np.ndarray
-            2D array of float x coordinates
+            1D array of float x coordinates
         y : np.ndarray
-            2D array of float y coordinates
+            1D array of float y coordinates
+        z : np.ndarray
+            1D array of float z coordinates
         """
+        if self.dims==3:
+            xx, yy, zz = self.get_rectmesh()
+            # modify to be even-R horizontal offset
+            xx[::2] += self.dx/4
+            xx[1::2] -= self.dx/4
+            if self.model_grid is not None:
+                xx, yy, zz = self.subset_floats_from_mask(xx, yy, zz)
+            return xx,yy,zz
+        else:
+            xx, yy = self.get_rectmesh()
+            # modify to be even-R horizontal offset 
+            xx[::2] += self.dx/4
+            xx[1::2] -= self.dx/4
 
-        xx, yy = self.get_rectmesh()
-        # modify to be even-R horizontal offset
-        xx[::2] += self.dx/4
-        xx[1::2] -= self.dx/4
-
-        if self.model_grid is not None:
-            xx, yy = self.subset_floats_from_mask(xx, yy)
-        return xx, yy
+            if self.model_grid is not None:
+                zz=np.ones(xx.size)*-0.5
+                xx, yy = self.subset_floats_from_mask(xx, yy, zz)
+            return xx, yy
```

Comment:
I think it's easier if `subset_floats_from_mask` always takes and returns the same number of arguments. It is (or should be) considered a private function, so we are allowed to change its arguments at will.